src/io/flutter/inspector/HeapDisplay.java

```diff
@@ -22,40 +24,96 @@
-  // The height of the drawable heap area.
-  private static final int HEAP_GRAPH_HEIGHT = 16;
+  public static AnAction createToolbarAction(Disposable parentDisposable, FlutterApp app) {
+    return new HeapDisplay.ToolbarComponentAction(parentDisposable, app);
+  }
 
-  // TODO(pq): consider basing this on available space or user configuration.
-  private static final boolean SHOW_HEAP_SUMMARY = false;
+  public static JPanel createJPanelView(Disposable parentDisposable, FlutterApp app) {
+    final JPanel panel = new JPanel(new BorderLayout());
+    panel.setBorder(IdeBorderFactory.createBorder(SideBorder.TOP));
+    panel.setPreferredSize(new Dimension(100, PANEL_HEIGHT));
+
+    final JBLabel rssLabel = new JBLabel();
+    rssLabel.setAlignmentY(Component.BOTTOM_ALIGNMENT);
+    rssLabel.setFont(UIUtil.getLabelFont(UIUtil.FontSize.SMALL));
+    rssLabel.setForeground(UIUtil.getLabelDisabledForeground());
+    rssLabel.setBorder(JBUI.Borders.empty(4));
+    final JBLabel heapLabel = new JBLabel("", SwingConstants.RIGHT);
+    heapLabel.setAlignmentY(Component.BOTTOM_ALIGNMENT);
+    heapLabel.setFont(UIUtil.getLabelFont(UIUtil.FontSize.SMALL));
+    heapLabel.setForeground(UIUtil.getLabelDisabledForeground());
+    heapLabel.setBorder(JBUI.Borders.empty(4));
+
+    final HeapState heapState = new HeapState(60 * 1000);
+    final HeapDisplay graph = new HeapDisplay(state -> {
+      rssLabel.setText(heapState.getRSSSummary());
+      heapLabel.setText(heapState.getHeapSummary());
+
+      SwingUtilities.invokeLater(rssLabel::repaint);
+      SwingUtilities.invokeLater(heapLabel::repaint);
+    });
+
+    graph.setLayout(new BoxLayout(graph, BoxLayout.X_AXIS));
+    graph.add(rssLabel);
+    graph.add(Box.createHorizontalGlue());
+    graph.add(heapLabel);
+
+    panel.add(graph, BorderLayout.CENTER);
+
+    final HeapListener listener = new HeapListener() {
+      @Override
+      public void handleIsolatesInfo(VM vm, List<IsolateObject> isolates) {
+        heapState.handleIsolatesInfo(vm, isolates);
+        graph.updateFrom(heapState);
+        SwingUtilities.invokeLater(panel::repaint);
+      }
 
-  @Nullable
-  private final SummaryCallback summaryCallback;
+      @Override
+      public void handleGCEvent(IsolateRef iIsolateRef, HeapSpace newHeapSpace, HeapSpace oldHeapSpace) {
+        heapState.handleGCEvent(iIsolateRef, newHeapSpace, oldHeapSpace);
+        graph.updateFrom(heapState);
+        SwingUtilities.invokeLater(panel::repaint);
+      }
+    };
+
+    assert app.getPerfService() != null;
+    app.getPerfService().addListener(listener);
+    Disposer.register(parentDisposable, () -> app.getPerfService().removeListener(listener));
+
+    return panel;
+  }
 
   public static class ToolbarComponentAction extends AnAction implements CustomComponentAction, HeapListener, Disposable {
+    // The width of the drawable heap area.
+    private static final int HEAP_GRAPH_WIDTH = 60;
```

Comment:
Maybe just rename this so you can delete the comment?  That is: `DRAWABLE_HEAP_WIDTH`?

Reply:
sgtm